tsconfig.json

```diff
@@ -1,7 +1,8 @@
 {
   "extends": "tsconfigs/nodejs-module",
   "compilerOptions": {
-    "outDir": "lib"
+    "outDir": "lib",
+    "skipLibCheck": true
```

Comment:
This may have been the reason I wasn't able to upgrade.

I wouldn't like to accept this. Can you find a different way to solve or work around the issue?

Reply:
The alternative are:

+ switch the project to ESM, which is a lot of work:

```json
{
"module": "ES2022",
"moduleResolution": "Node16",
}  
```

And "type": "module" in package.json 
 
+ wrap the library switched to ESM to CommonJS exports (@typescript-eslint/eslint-plugin)

Reply:
Tell me more about this wrapping, please.

Reply:
> 
> * overload locally the culprit library type (@typescript-eslint/eslint-plugin)
> 
> See https://github.com/typescript-eslint/typescript-eslint/issues/7284

I do not see how to make it works in a very hackish way, the original culprit types are already an ugly hack: 

- @typescript-eslint/eslint-plugin/index.d.ts:1:31
- @typescript-eslint/eslint-plugin/rules.d.ts::38:33

Maybe by redefining the imported types locally or with patch-package to fix the types declaration for the TS configuration: self-contained without import. It's still an ugly hack requiring quite of work.  

Code defining custom rules with @typescript-eslint/eslint-plugin needs to switch to "Bundler" module resolution. And the amount of work to do it properly is less than trying to find a working workaround not requiring "skipLibCheck" given the repo TS code surface. 

So, maybe accepts temporary the current solution to support v6 and prepares a PR for the proper v6 support by switching to ESM and module resolution to "Bundler". It's easier because there's already ESM wrapper to deal with dependencies using CommonJS.

Reply:
The `patch-package` approach sounds reasonable. Also, progress is being made upstream eslint-config-standard. If that turns into an ESM, then we would be able to turn this package ESM, as well..